test/deploy/os_ova_install.py

```diff
@@ -51,64 +63,78 @@ def test01_install_ova_template(self):
-                             "--net:'ADMIN'='VM Network' "
-                             "--net:'CONTROL'='Control Network' "
-                             "--net:'PDU'='PDU Network' "
-                             "--name='ora-stack-" + fit_common.ARGS_LIST['stack'] + "' "
-                             "--noSSLVerify "
-                             + ovafile
-                             + " vi://" + uname + ":" + passwd + "@"
-                             + fit_common.ARGS_LIST['hyper'], shell=True)
-        # Check for successful completion
-        if rc > 0:
-            print 'OVA installer failed at host: ' + fit_common.ARGS_LIST['hyper'] + "Exiting..."
-            sys.exit(255)
-
-        # Wait for VM to settle
-        fit_common.countdown(30)
-        # Install MAC address by editing OVA .vmx file, then startup VM
-        esxi_command = "export fullpath=`find vmfs -name ora-stack-" + fit_common.ARGS_LIST['stack'] + "*.vmx`;" \
-                       "for file in $fullpath;" \
-                       "do " \
-                       "export editline=`cat $file |grep \\\'ethernet0.generatedAddress =\\\'`;" \
-                       "export editcmd=\\\'/\\\'$editline\\\'\/ c\\\ethernet0.address = \\\"" + fit_common.STACK_CONFIG[fit_common.ARGS_LIST['stack']]['ovamac'] + "\\\"\\\';" \
-                       "sed -i \\\"$editcmd\\\" $file;" \
-                       "sed -i \\\'/ethernet0.addressType = \\\"vpx\\\"/ c\\\ethernet0.addressType = \\\"static\\\"\\\' $file;" \
-                       "sed -i \\\'/ethernet0.addressType = \\\"generated\\\"/ c\\\ethernet0.addressType = \\\"static\\\"\\\' $file;" \
-                       "done;" \
-                       "sleep 5;" \
-                       "export vmidstring=`vim-cmd vmsvc/getallvms |grep ora-stack-" + fit_common.ARGS_LIST['stack'] + "`;" \
-                       "for vmid in $vmidstring;" \
-                       "do " \
-                       "vim-cmd vmsvc/power.on $vmid;" \
-                       "exit $?;" \
-                       "done;"
-
-        (command_output, exitstatus) = \
-            fit_common.pexpect.run(
-                            "ssh -q -o StrictHostKeyChecking=no -t " + uname + "@"
-                            + fit_common.ARGS_LIST['hyper'] + " " + esxi_command,
-                            withexitstatus=1,
-                            events={"assword": passwd + "\n"},
-                            timeout=20, logfile=sys.stdout)
-        if exitstatus > 0:
-            print "MAC address processing failed. Exiting..."
-            sys.exit(255)
-
-        # Poll the OVA via ping
-        for dummy in range(0, 30):
-            rc = subprocess.call("ping -c 1 -w 5 " + fit_common.ARGS_LIST['ora'], shell=True)
-            if rc == 0:
-                break
+        for vm in range(0, numvms):
+            print '**** Deploying OVA file on hypervisor ' + fit_common.ARGS_LIST['hyper']
```

Comment:
Please create a sub function to handle the process about deploying OVA file for one VM (all codes inside this `for` loop), so the code will look more clean.